### ISSUE TITLE
Template-Based Data Augmentation comprehension issue

### ISSUE THREAD

**Issue by deeplearnerguy** [ANTsX/ANTsRNet#24]
Hi,

I'm trying to implement data augmentation for U-net training on MRI images using template-based augmentation method described in "Convolutional Neural Networks with Template-Based Data Augmentation for Functional Lung Image Quantification" (https://www.ncbi.nlm.nih.gov/pubmed/30195415). 

What i'm trying to understand is how the new data is created. The following formula is used : 

Snew = Ssource(φinvtarget(φsource)). 

My understanding of the formula is : Ssource will be mapped to the template space, and then mapped with the inverse φ to Starget, this result is Snew. In my comprehension, Ssource will be mapped to the target space, and if φ is a perfect mapping, the result Snew will be exactly similar to Starget.

My question is, is Snew exactly similar to Starget ?
If yes, how the data could be augmented ?

Thanks for your answer,

Regards

**Comment by ntustison** [comment]
Unless under artificially contrived conditions, ``Snew`` will not be exactly like ``Starget``.  You'll have structures which don't correspond across subjects (e.g., tertiary sulci in the brain) and the relative intensities will be different.  Also, the way we'll typically do template data augmentation is create a template from the *masks* of the structures of interest and simply map between subjects based on masks only.  That way the overall shape is within the range of the population but the voxels both internal and external to the mask boundaries will vary.  That's what we did for the paper for lung proton MRI.  Here are the [training scripts](https://github.com/ANTsXNet/ProtonMRILungSegmentation/tree/master/Scripts/Training) if you're interested although they're a bit dated since R6 objects have been replaced by functions.  [Here](https://github.com/ANTsXNet/HumanCTLungSegmentation/blob/master/Scripts/Training/unetBatchGenerator.R) are a set of scripts using the more recent keras function style.
